src/hyperion/experiment_plans/oav_snapshot_plan.py

```diff
@@ -1,28 +1,55 @@
-import dataclasses
 from datetime import datetime
+from typing import Protocol
 
-from blueapi.core import BlueskyContext, MsgGenerator
+from blueapi.core import MsgGenerator
 from bluesky import plan_stubs as bps
+from dodal.devices.aperturescatterguard import AperturePositions, ApertureScatterguard
+from dodal.devices.backlight import Backlight, BacklightPosition
 from dodal.devices.oav.oav_detector import OAV
 from dodal.devices.oav.oav_parameters import OAVParameters
 from dodal.devices.smargon import Smargon
 
 from hyperion.device_setup_plans.setup_oav import setup_general_oav_params
 from hyperion.parameters.components import WithSnapshot
 from hyperion.parameters.constants import DocDescriptorNames
-from hyperion.utils.context import device_composite_from_context
 
+OAV_SNAPSHOT_SETUP_GROUP = "oav_snapshot_setup"
 OAV_SNAPSHOT_GROUP = "oav_snapshot_group"
 
 
-@dataclasses.dataclass
-class OavSnapshotComposite:
+class OavSnapshotComposite(Protocol):
     smargon: Smargon
     oav: OAV
+    aperture_scatterguard: ApertureScatterguard
+    backlight: Backlight
 
 
-def create_devices(context: BlueskyContext) -> OavSnapshotComposite:
-    return device_composite_from_context(context, OavSnapshotComposite)  # type: ignore
+def setup_oav_snapshot_plan(composite: OavSnapshotComposite, parameters: WithSnapshot):
+    if not parameters.take_snapshots:
+        return
+
+    yield from bps.abs_set(
+        composite.backlight, BacklightPosition.IN, group=OAV_SNAPSHOT_SETUP_GROUP
+    )
+    yield from bps.abs_set(
+        composite.aperture_scatterguard,
+        AperturePositions.ROBOT_LOAD,
+        group=OAV_SNAPSHOT_SETUP_GROUP,
+    )
+
+
+def oav_snapshot_plan(
+    composite: OavSnapshotComposite,
+    parameters: WithSnapshot,
+    oav_parameters: OAVParameters,
+    wait: bool = True,
+) -> MsgGenerator:
+    if not parameters.take_snapshots:
+        return
+    yield from bps.wait(group=OAV_SNAPSHOT_SETUP_GROUP)
+    yield from _setup_oav(composite, parameters, oav_parameters)
+    for omega in parameters.snapshot_omegas_deg or []:
```

Comment:
I did what you suggested but then was reminded why I didn't - pyright complains that it can still be None, which you still need in the type hint in order to allow the parameter to be omitted rather than left empty in the json. So I figured might as well leave the check in there.